ctapipe/calib/camera/r1.py

```diff
@@ -9,24 +9,51 @@
 As the R1 calibration is camera specific, each camera (and seperately the MC)
 requires their own calibrator class with inherits from `CameraR1Calibrator`.
 `HessioR1Calibrator` is the calibrator for the MC data obtained from readhess.
-Through the use of `CameraR1CalibratorFactory`, the correct
+Through the use of `camera_r1_calibrator_for_eventsource()`, the correct
 `CameraR1Calibrator` can be obtained based on the origin (MC/Camera format)
 of the data.
 """
 from abc import abstractmethod
 import numpy as np
-from ...core import Component, Factory
+
+from ...core import Component
 from ...core.traits import Unicode
-from ...io import EventSource
 
 __all__ = [
     'NullR1Calibrator',
     'HESSIOR1Calibrator',
     'TargetIOR1Calibrator',
-    'CameraR1CalibratorFactory'
+    'CameraR1Calibrator',
+    'camera_r1_calibrator_for_eventsource',
 ]
 
 
+def camera_r1_calibrator_for_eventsource(eventsource, **kwargs):
```

Comment:
Shouldn't this be a constructor for CameraR1Calibrator, following the convention that factories are really just specialized constructors of the base class?  e.g. `cal = CameraR1Calibrator.for_event_source(source)`

Reply:
See my comment below.